libfwupdplugin/fu-udev-device.c

```diff
@@ -1385,34 +1287,9 @@ fu_udev_device_open(FuDevice *device, GError **error)
 static gboolean
 fu_udev_device_rescan(FuDevice *device, GError **error)
 {
-#ifdef HAVE_GUDEV
-	FuUdevDevice *self = FU_UDEV_DEVICE(device);
-	FuUdevDevicePrivate *priv = GET_PRIVATE(self);
-	const gchar *sysfs_path;
-	g_autoptr(GUdevClient) udev_client = g_udev_client_new(NULL); /* nocheck:blocked */
-	g_autoptr(GUdevDevice) udev_device = NULL;
-
-	/* never set */
-	if (priv->udev_device == NULL) {
-		g_set_error_literal(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_INTERNAL,
-				    "rescan with no previous device");
-		return FALSE;
-	}
-	sysfs_path = g_udev_device_get_sysfs_path(priv->udev_device);
-	udev_device = g_udev_client_query_by_sysfs_path(udev_client, sysfs_path);
-	if (udev_device == NULL) {
-		g_set_error(error,
-			    FWUPD_ERROR,
-			    FWUPD_ERROR_INTERNAL,
-			    "rescan could not find device %s",
-			    sysfs_path);
-		return FALSE;
-	}
-	fu_udev_device_set_dev(self, udev_device);
+	if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_EMULATED))
+		return TRUE;
```

Comment:
Hmm, so what do you want to do about emulation as it pertains to a sysfs file changing as part of a flash process?

Reply:
This made my head spin a little at first too -- essentially the old code was saying "the GUdevDevice that backs the FuUdevDevice might have changed" as GUdevDevice is a mostly immutable snapshot of a udev event. Now that FuUdevEvent actually models the sysfs device itself it no longer has to be invalidated. The contents of the sysfs might have changed (hence why we invalidate the probe data), but for emulations that's never going to happen (as it's the same device)  and we don't actually want to replay the probe out of order.